src/DynamoCoreWpf/ViewModels/Watch3D/Shaders/Custom.fx

```diff
@@ -139,13 +139,13 @@ PSInputCustom VShaderCustom(VSInputCustom input)
 	PSInputCustom output;
 
 	bool isSelected = input.customParams.x;
-	bool requiresPerVertexColoration = input.customParams.y;
+	bool requiresPerVertexColoration = input.customParams.z;
 
 	float4 inputp;
 	if (isSelected || requiresPerVertexColoration)
 	{
 		// Nudge the vertex out slightly along its normal.
-		float3 nudge = input.n * 0.0001;
+		float3 nudge = normalize(input.n) * 0.0001;
```

Comment:
@ellensi @kirsch This does not seem to work well with selected display geometries (Display.byGeometryColor)?
Steps:
1. Start D4R 2.0.0.4196, default Automatic
2. Populate ColorByARGB, Sphere.ByCenterPointRadius and Display.ByGeometryColor
3. Hook them all up, with Red==255, G == 0, B==0 and A==255
// sphere is unchanged from gray: not red: INCORRECT

If you turn off the Sphere's Preview setting, then the color is rendered correctly.